src/praisonai/tests/unit/test_sequential_tool_calling.py

```
@@ -0,0 +1,413 @@
+#!/usr/bin/env python3
+"""
+Comprehensive mock tests for sequential tool calling functionality in PraisonAI.
+This test suite covers various scenarios of sequential tool execution.
+"""
+
+import sys
+import os
+import pytest
+import json
```

Ruff (0.11.9)

`json` imported but unused

Remove unused import: `json`

(F401)